rpcs3/rpcs3qt/user_manager_dialog.cpp

```diff
@@ -0,0 +1,375 @@
+		// but I can't find proof of this always being true.
+		for (const auto& userFolder : fs::dir(base_dir))
+		{
+			if (!userFolder.is_directory)
+			{
+				continue;
+			}
+
+			// Is the folder name exactly 8 all-numerical characters long?
+			// We use strtol to find any non-numeric characters in folder name.
+			char* nonNumericChar;
+			std::strtol(userFolder.name.c_str(), &nonNumericChar, 10);
+			if (userFolder.name.length() != 8 || *nonNumericChar != '\0')
+			{
+				continue;
+			}
+
+			// Does the localusername file exist?
+			if (!fs::is_file(fmt::format("%s/%s/localusername", base_dir, userFolder.name)))
+			{
+				continue;
+			}
+
+			UserAccount* user_entry = new UserAccount(userFolder.name);
+			user_list.emplace_back(user_entry);
+		}
+		return user_list;
+	}
+}
+
+user_manager_dialog::user_manager_dialog(std::shared_ptr<gui_settings> gui_settings, std::shared_ptr<emu_settings> emu_settings, const std::string& dir, QWidget* parent)
+	: QDialog(parent), m_user_list(), m_dir(dir), m_sort_column(1), m_sort_ascending(true), m_gui_settings(gui_settings), m_emu_settings(emu_settings)
+{
+	setWindowTitle(tr("User Manager"));
+	setMinimumSize(QSize(400, 400));
+	setModal(true);
+
+	m_emu_settings->LoadSettings();
+	Init(dir);
+}
+
+void user_manager_dialog::Init(const std::string& dir)
+{
+	// Table
+	m_table = new QTableWidget(this);
+
+	//m_table->setItemDelegate(new table_item_delegate(this)); // to get rid of cell selection rectangles include "table_item_delegate.h"
+	m_table->setSelectionMode(QAbstractItemView::SelectionMode::SingleSelection);
+	m_table->setSelectionBehavior(QAbstractItemView::SelectRows);
+	m_table->setContextMenuPolicy(Qt::CustomContextMenu);
+	m_table->setColumnCount(2);
+	m_table->setHorizontalHeaderLabels(QStringList() << tr("User ID") << tr("User Name"));
+
+	QPushButton* push_remove_user = new QPushButton(tr("Delete User"), this);
+	QPushButton* push_create_user = new QPushButton(tr("Create User"), this);
+	QPushButton* push_login_user = new QPushButton(tr("Log In User"), this);
+	QPushButton* push_rename_user = new QPushButton(tr("Rename User"), this);
+
+	QPushButton* push_close = new QPushButton(tr("&Close"), this);
+	push_close->setAutoDefault(true);
+
+	// Button Layout
+	QHBoxLayout* hbox_buttons = new QHBoxLayout();
+	hbox_buttons->addWidget(push_create_user);
+	hbox_buttons->addWidget(push_login_user);
+	hbox_buttons->addWidget(push_rename_user);
+	hbox_buttons->addWidget(push_remove_user);
+	hbox_buttons->addStretch();
+	hbox_buttons->addWidget(push_close);
+
+	// main layout
+	QVBoxLayout* vbox_main = new QVBoxLayout();
+	vbox_main->setAlignment(Qt::AlignCenter);
+	vbox_main->addWidget(m_table);
+	vbox_main->addLayout(hbox_buttons);
+	setLayout(vbox_main);
+
+	m_selected_user = m_emu_settings->GetSetting(emu_settings::SelectedUser);
+	UpdateTable();
+
+	QByteArray geometry = m_gui_settings->GetValue(gui::um_geometry).toByteArray();
```

Comment:
I think you can just say: 
`restoreGeometry(m_gui_settings->GetValue(gui::um_geometry).toByteArray());`
since that will just return on invalid input.
I just changed that on the rest of those occurances. there would be no need for the if () anymore